From cee3188c1abaa5b222c57b87cc94064e56921db8 Mon Sep 17 00:00:00 2001
From: Vadim Zeitlin <vadim wxwidgets.org>
Date: Mon, 29 Feb 2016 18:47:30 +0100
Subject: [PATCH] Don't connect handler for wxEVT_TEXT_ENTER in wxComboCtrl
 unnecessarily

Connecting to wxEVT_TEXT_ENTER is not necessary if wxTE_PROCESS_ENTER style is
not used was always useless and is actively harmful since the changes of
5591a2009394c9bae43bcd2fa08f2fb51391cbc1 as it results in an assert failure
now.

Just don't do it unless we really want, and will get, these events.
---
diff --git a/src/common/combocmn.cpp b/src/common/combocmn.cpp
index 4ab9431..d451d68 100644
--- src/common/combocmn.cpp.orig
+++ src/common/combocmn.cpp
@@ -1100,9 +1100,12 @@ wxComboCtrlBase::CreateTextCtrl(int style)
         m_text->Connect(id, wxEVT_TEXT,
                         wxCommandEventHandler(wxComboCtrlBase::OnTextCtrlEvent),
                         NULL, this);
-        m_text->Connect(id, wxEVT_TEXT_ENTER,
-                        wxCommandEventHandler(wxComboCtrlBase::OnTextCtrlEvent),
-                        NULL, this);
+        if ( style & wxTE_PROCESS_ENTER )
+        {
+            m_text->Connect(id, wxEVT_TEXT_ENTER,
+                            wxCommandEventHandler(wxComboCtrlBase::OnTextCtrlEvent),
+                            NULL, this);
+        }
 
         m_text->SetHint(m_hintText);
     }

From ee486dba32d02c744ae4007940f41a5b24b8c574 Mon Sep 17 00:00:00 2001
From: Vadim Zeitlin <vadim wxwidgets.org>
Date: Mon, 29 Feb 2016 18:05:02 +0100
Subject: [PATCH] Fix build error in wxOSX clang C++11 build for i386

When building for i386 in C++11 mode clang warns about narrowing conversion
from double to CGFloat (== float) in the initialization.

Fix it by using explicit suffixes for the float literals.
---
diff --git a/src/osx/carbon/graphics.cpp b/src/osx/carbon/graphics.cpp
index 091c2c2..2d9cb96 100644
--- src/osx/carbon/graphics.cpp.orig
+++ src/osx/carbon/graphics.cpp
@@ -2276,7 +2276,7 @@ void wxMacCoreGraphicsContext::DoDrawText( const wxString &str, wxDouble x, wxDo
     {
         CGFloat width = CTLineGetTypographicBounds(line, NULL, NULL, NULL);
         CGFloat height = CTFontGetXHeight( font );
-        CGPoint points[] = { {0.0, height * 0.6},  {width, height * 0.6} };
+        CGPoint points[] = { {0.0, height * 0.6f},  {width, height * 0.6f} };
         CGContextSetStrokeColorWithColor(m_cgContext, col);
         CGContextSetShouldAntialias(m_cgContext, false);
         CGContextSetLineWidth(m_cgContext, 1.0);

From 173ecd77c4280e48541c33bdfe499985852935ba Mon Sep 17 00:00:00 2001
From: Vadim Zeitlin <vadim wxwidgets.org>
Date: Mon, 29 Feb 2016 16:12:53 +0100
Subject: [PATCH] Fix clang -Winconsistent-missing-override in wxTextCtrl

Add missing wxOVERRIDE for OnDynamicBind().
---
diff --git a/include/wx/textctrl.h b/include/wx/textctrl.h
index cd7a815..3414095 100644
--- include/wx/textctrl.h.orig
+++ include/wx/textctrl.h
@@ -746,7 +746,7 @@ class WXDLLIMPEXP_CORE wxTextCtrlBase : public wxControl,
     // Override wxEvtHandler method to check for a common problem of binding
     // wxEVT_TEXT_ENTER to a control without wxTE_PROCESS_ENTER style, which is
     // never going to work.
-    virtual bool OnDynamicBind(wxDynamicEventTableEntry& entry);
+    virtual bool OnDynamicBind(wxDynamicEventTableEntry& entry) wxOVERRIDE;
 
     // override streambuf method
 #if wxHAS_TEXT_WINDOW_STREAM
